### PR TITLE
pre-commit: Add codespell and pyupgrade

### PR DESCRIPTION
**Pull request by cclauss**
This GitHub Action contains a useful set of tests on its own.

But also, putting configs for these tools into `setup.cfg` and then ensuring that they properly convert over to `pyproject.toml` would help a lot of projects.

### CODE REVIEW

**Comment by abravalheri** [comment]
Hi @cclauss, thank you very much for the contribution.

I suspect that very similar checks are being executed already, but using a different CI system (Cirrus CI): https://github.com/abravalheri/ini2toml/blob/e4116df84bd4e27c8d46fc949389e1500f39e608/.cirrus.yml#L194

Is there anyway we can unify these settings there?

Most of the times I try to make the same task that I use in the CI also available to run locally.

I am relying on `pre-commit` right now for that ([config file](https://github.com/abravalheri/ini2toml/blob/main/.pre-commit-config.yaml)), and I have some plumbing to run them on the CI via tox ([config file](https://github.com/abravalheri/ini2toml/blob/e4116df84bd4e27c8d46fc949389e1500f39e608/tox.ini#L28)).

**Comment by cclauss** [comment]
Perfect.  Let's do it via pre-commit.

**Comment by abravalheri** [comment]
Perfect!

Once it is in pre-commit, this should already affect all the CI environments ([GitHub actions for example](https://github.com/abravalheri/ini2toml/blob/e4116df84bd4e27c8d46fc949389e1500f39e608/.github/workflows/ci.yml#L33))

File: src/ini2toml/types.py [hunk 1]
```diff
@@ -17,7 +17,7 @@
 if sys.version_info <= (3, 8):  # pragma: no cover
     # TODO: Import directly when `python_requires = >= 3.8`
     if TYPE_CHECKING:
-        from typing_extensions import Protocol
```
Comment (cclauss) [inline src/ini2toml/types.py:20]: What is the minimum Python supported?  Pyupgrade made this change because I said `--py38-plus`

Comment (abravalheri) [reply src/ini2toml/types.py:20]: Hi @cclauss, I am committed to follow Python's EoL. So this would mean supporting Python 3.7 until 2023-06-27.

**Comment by abravalheri** [comment]
Thank you very much @cclauss

**Pull request merged**